bokehjs/src/coffee/models/formatters/datetime_tick_formatter.coffee

```diff
@@ -77,26 +61,26 @@ class DatetimeTickFormatter extends TickFormatter.Model
     'hours':        ['%Hh', '%H:%M']
     'days':         ['%m/%d', '%a%d']
     'months':       ['%m/%Y', '%b%y']
-    # TODO: the functions here currently aren't serializable
-    # and so can't be set from the python client. Although,
-    # I'm not at all sure what purpose these functions have.
-    'years':        ['%Y', _two_digit_year, _four_digit_year]
+    'years':        ['%Y']
   }
 
   # Whether or not to strip the leading zeros on tick labels.
   strip_leading_zeros: true
 
   initialize: (attrs, options) ->
     super(attrs, options)
+    @formats = _.extend({}, @_formats, @get("formats"))
```

Comment:
I presume this is intentional, but let me reiterate. This can be viewed as `@set("formats", _.extend(...))`, so you are modifying user's input and (when using server) pushing changes to the client. I would say a computed property should have been defined here (e.g. `computed_formats`). In this case it may be harmless to leave it as-is. However, in general I think we shouldn't modify user's input like this. This is specifically visible in `Plot.renderers` where we mix all kinds of stuff and in the end, it requires special serialization code to make it work (among other issues).